x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality_panel/data_quality_details/indices_details/pattern/index_check_flyout/historical_results/historical_results_list/historical_result/historical_check_fields/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { IncompatibleTab } from '../../../../incompatible_tab';
+import { SameFamilyTab } from '../../../../same_family_tab';
+import { CheckFieldsTabs } from '../../../../check_fields_tabs';
+import { StyledHistoricalResultsCheckFieldsButtonGroup } from '../styles';
+
+export interface Props {
+  indexName: string;
+  historicalResult: NonLegacyHistoricalResult;
+}
+
+const HistoricalCheckFieldsComponent: React.FC<Props> = ({ indexName, historicalResult }) => {
+  const { incompatibleMappingsFields, incompatibleValuesFields, sameFamilyFields } =
+    getIncompatibleAndSameFamilyFieldsFromHistoricalResult(historicalResult);
+
+  const {
+    docsCount,
+    ilmPhase,
+    sizeInBytes,
+    incompatibleFieldCount,
+    sameFamilyFieldCount,
+    ecsFieldCount,
+    customFieldCount,
+    totalFieldCount,
+  } = historicalResult;
+
+  const tabs = useMemo(
+    () => [
+      {
+        id: INCOMPATIBLE_TAB_ID,
+        name: INCOMPATIBLE_FIELDS,
+        badgeColor: getIncompatibleStatBadgeColor(incompatibleFieldCount),
+        badgeCount: incompatibleFieldCount,
+        content: (
+          <IncompatibleTab
+            docsCount={docsCount}
+            ilmPhase={ilmPhase}
+            indexName={indexName}
+            incompatibleMappingsFields={incompatibleMappingsFields}
+            incompatibleValuesFields={incompatibleValuesFields}
+            sameFamilyFieldsCount={sameFamilyFieldCount}
+            ecsCompliantFieldsCount={ecsFieldCount}
+            customFieldsCount={customFieldCount}
+            allFieldsCount={totalFieldCount}
+            sizeInBytes={sizeInBytes}
+            hasStickyActions={false}
+          />
+        ),
+      },
+      {
+        id: SAME_FAMILY_TAB_ID,
+        name: SAME_FAMILY,
+        badgeColor: getIncompatibleStatBadgeColor(sameFamilyFieldCount),
+        badgeCount: sameFamilyFieldCount,
+        content: (
+          <SameFamilyTab
+            docsCount={docsCount}
+            ilmPhase={ilmPhase}
+            indexName={indexName}
+            sameFamilyFields={sameFamilyFields}
+            incompatibleFieldsCount={incompatibleFieldCount}
+            ecsCompliantFieldsCount={ecsFieldCount}
+            customFieldsCount={customFieldCount}
+            allFieldsCount={totalFieldCount}
+            sizeInBytes={sizeInBytes}
+            hasStickyActions={false}
+          />
+        ),
+      },
+    ],
+    [
+      customFieldCount,
+      docsCount,
+      ecsFieldCount,
+      ilmPhase,
+      incompatibleFieldCount,
+      incompatibleMappingsFields,
+      incompatibleValuesFields,
+      indexName,
+      sameFamilyFieldCount,
+      sameFamilyFields,
+      sizeInBytes,
+      totalFieldCount,
+    ]
+  );
+
+  return (
+    <div data-test-subj="historicalCheckFields">
+      <CheckFieldsTabs
+        tabs={tabs}
+        renderButtonGroup={(props) => <StyledHistoricalResultsCheckFieldsButtonGroup {...props} />}
```

Comment:
I don't quite understand why do we need this `renderButtonGroup` prop, can't we just render `StyledHistoricalResultsCheckFieldsButtonGroup` inside the `CheckFieldsTabs`?

Reply:
I there are 3 places now where we use the new `<CheckFieldTabs />` and in in 2 we use different ButtonGroups (configs + style)

So this renderProp allows to put in the needed version of ButtonGroup

Reply:
Oh yes, my bad, I missed the one using `StyledButtonGroup`. 
A more self-contained approach could be passing a `widthMode='full' | 'half'` prop or something like that instead, so all the different available versions are encapsulated inside the `CheckFieldsTabs` implementation, but maybe I am missing something. The current approach is fine for me.

Reply:
yes but that would bloat the CheckFieldsTabs with having to maintain the variations of styled nodes. It's imho arguably more clean when the consumer maintains its own version of styled components.